src/charm_state.py

```diff
@@ -16,6 +18,37 @@
 )
 
 
+class ProxyConfig(BaseModel):
```

Comment:
What is the reason for not making this part of the `CharmState` class?

Reply:
None in particular. Are you suggesting that to improve consistency with other charms?

Reply:
Well that and why not just make it part of the charm state?

Reply:
@jdkandersson @gregory-schiano Did you came to a decision about that as mentioned here?
https://github.com/canonical/flask-k8s-operator/pull/33#discussion_r1271782149

Reply:
Not yet, I'll discuss with gregory today

Reply:
Actually we did discuss it, it should go into the state

Reply:
Done